inc/Integrations/SalesforceD2C/Auth/SalesforceD2CAuth.php

```diff
@@ -0,0 +1,200 @@
+<?php declare(strict_types = 1);
+
+namespace RemoteDataBlocks\Integrations\SalesforceD2C\Auth;
+
+use WP_Error;
+
+/**
+ * Salesforce D2C Auth class.
+ *
+ * This class is used to authenticate with Salesforce D2C using a client ID and secret.
+ */
+class SalesforceD2CAuth {
+
+	/**
+	 * Generate a token from a client ID and secret, or use an existing token if available.
+	 *
+	 * @param string $endpoint The endpoint prefix URL for the data source,
+	 * @param string $client_id The client ID (a version 4 UUID).
+	 * @param string $client_secret The client secret.
+	 * @return string|WP_Error The token or an error.
+	 */
+	public static function generate_token(
+		string $endpoint,
+		string $client_id,
+		string $client_secret
+	): string|WP_Error {
+		return self::get_saved_access_token( $client_id ) ?? self::get_token_using_client_credentials( $client_id, $client_secret, $endpoint );
+	}
+
+	/**
+	 * Get the webstores using the given endpoint, and token.
+	 *
+	 * @param string $endpoint The endpoint prefix URL for the data source.
+	 * @param string $token The token.
+	 * @return array|WP_Error The webstores or an error.
+	 */
+	public static function get_webstores(
```

Comment:
This is here for the same reason that the AuthController has this method - it's just for token verification. For the checkout API we will likely need another token so it won't use this. I also am reluctant to reveal the token outside of the server just for security sake.